libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -104,6 +104,21 @@ class YulUtilFunctions
 	std::string overflowCheckedUIntSubFunction();
 
 	std::string arrayLengthFunction(ArrayType const& _type);
+
+	/// @returns the name of a function that resizes an array
+	/// signature: (array, newLen)
+	std::string resizeDynamicArrayFunction(ArrayType const& _type);
+
+	/// @returns the name of a function that will clear the storage area given
+	/// by the start and end parameters
+	/// signature: (start, end)
+	std::string clearStorageRangeFunction(Type const& _type);
+
+	/// Returns the name of a function that will convert a given length to the
+	/// size it will require
```

Comment:
Please clarify: It probably only includes the "head size" in memory and storage. What about calldata?

Reply:
Eventually it will be able to handle calldata, too. In this PR i focused mainly on `storage`. Any memory/calldata code is just a by-product while I was working on that (because it was conveniently at the same place for example). 

I plan to address anything non-storage specific in the next PR.

Though, in this particular case, supporting calldata seems to require just one line different to memory:
From the original evm code:

```
			if (_arrayType.location() == DataLocation::Memory)
				m_context << _arrayType.baseType()->memoryHeadSize();
			else
				m_context << _arrayType.baseType()->calldataEncodedSize();
```

Not sure what you mean by "head size", though

Reply:
head size: memoryHeadSize. Not sure what would be a better name. For storage, it would not include the dynamic part of arrays, but it would include the static part. Perhaps it is less confusing to not be that specific after all.